lib/utils.js

```diff
@@ -161,11 +161,20 @@ module.exports.mapToMac = function (options) {
 module.exports.actionJackerDecorator = function (emitter, options, fn, mapper) {
   options = clone(options);
   fn = fn || function (err, data) {};
-  return function (err, data) {
-    fn.apply(emitter, [err, data]);
-    if (err || !mapper || !data) return;
-
-    var key = mapper(data);
+  return function (err, data) {
+
+    var resultantData = data;

```

Comment:
Sometimes `data` can be `undefined` and will throw errors if we try to use string methods on it so we just default the result to the `data` and if it is available, do some work on it and return `resultantData` no matter what happened.